### DISCUSSION TITLE
Connection Pool

### DISCUSSION THREAD

**Discussion by Vidhya-RajaMani** [jtv/libpqxx#799] [Q&A]
Is there any feature to implement the connection Pool using this library?

**Comment by tt4g** [comment]
Duplicate #699

There are pooling libraries such as pgpool and pgbouncer, which are not libraries for libpqxx.

**Reply by Vidhya-RajaMani** [reply]
Thank you for your response

**Comment by jtv** [comment]
I suppose it would be _possible_ to write a connection pooling library as an add-on to libpqxx.  But in all honesty, the thought of reusing connections always makes me a little nervous.  How do you ensure that all of its state is reset to a known state, on both sides of the connection?

**Reply by RedTrait** [reply]
I feel that the connection pool could be provided as a battery library, separate it from the library of libpqxx, but I am now trying to use libpqxx to connect to pgpool-II, and in my code I have the connection to pgpool-II as an object pool

**Reply by jtv** [reply]
In principle it should be possible...  Although I have no idea how we'd reliably reset a connection's state for reuse.  There's both client-side and server-side state to be considered.

So yes, an external pool is probably safer.  Which is not to say that I wouldn't be interested if someone built such a library!

**Reply by RedTrait** [reply]
I think the first thing to do this is probably to choose an asynchronous runtime. Will you choose libuv? I think this is a very good choice.

**Reply by jtv** [reply]
To be clear, I would not have time to develop it, so'I shouldn't be choosing!

But if you're asking me for my _preferences,_ I would avoid making that choice if possible, and see whether it could be done with nonblocking sockets so that the user can make their own choices based on what the application needs.

**Reply by RedTrait** [reply]
thank you answer me

**Comment by jtv** [comment]
I suppose we're answered this one, really.
